src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/CloneManager.cs

```diff
@@ -139,7 +141,19 @@ protected async Task<NativePath> PrepareCloneInternal(string remoteUri, string d
         }
         else
         {
-            _logger.LogDebug("Clone of {repo} found in {clonePath}", remoteUri, clonePath);
+            _logger.LogDebug("Clone of {repo} found in {clonePath}. Preparing for use...", remoteUri, clonePath);
+
+            // We make sure the clone is clean and we re-clone if it's unusable
+            if (performCleanup)
+            {
+                var result = await _localGitRepo.RunGitCommandAsync(clonePath, ["reset", "--hard"], cancellationToken);
+                if (!result.Succeeded)
+                {
+                    _logger.LogWarning("Failed to clean up {clonePath}, re-cloning", clonePath);
+                    _fileSystem.DeleteDirectory(clonePath, recursive: true);
+                    return await PrepareCloneInternal(remoteUri, dirName, performCleanup: true, cancellationToken);
```

Comment:
If it really clones again, it will no go this route anymore but rather through line 139 so it does not really matter what we put there. But I assumed `true` is better in case of whatever other new case in the future.